### PR TITLE
Write roman ts and repeats

### PR DESCRIPTION
**Pull request by MarkGotham**
Write complement to #1378 along with a bit of clean up

### CODE REVIEW

File: music21/romanText/writeRoman.py [hunk 1]
```diff
@@ -235,17 +235,35 @@ def prepSequentialListOfLines(self):
         adding this to the (already prepared) metadata preamble ready for printing.
 
         >>> p = stream.Part()
-        >>> m = stream.Measure()
+        >>> m = stream.Measure(number=1)
         >>> m.insert(0, meter.TimeSignature('4/4'))
         >>> m.insert(0, roman.RomanNumeral('V', 'G'))
-        >>> p.insert(0, m)
+        >>> p.insert(m)
```
Comment (mscuthbert) [inline music21/romanText/writeRoman.py:241]: -1; I'm deemphasizing using insert without an offset.  Unnecessary change.

File: music21/romanText/writeRoman.py [hunk 2]
```diff
@@ -235,17 +235,35 @@ def prepSequentialListOfLines(self):
...
         >>> testCase = romanText.writeRoman.RnWriter(p)
         >>> testCase.combinedList[-1]  # Last entry, after the metadata
-        'm0 G: V'
+        'm1 G: V'
+
+        This follows the wider rntxt syntax in supporting
+        Time Signature (:class:`~music21.meter.TimeSignature`) changes and
+        Repeats marks (:class:`~music21.bar.Repeat`)
+        but only (currently) between measures.
+        Mid-measure repeats are an open issue (much wider than rntxt).
```
Comment (mscuthbert) [inline music21/romanText/writeRoman.py:250]: remove editorial

File: music21/romanText/writeRoman.py [hunk 3]
```diff
@@ -235,17 +235,35 @@ def prepSequentialListOfLines(self):
...
+
+        >>> m2 = stream.Measure(number=2)
```
Comment (mscuthbert) [inline music21/romanText/writeRoman.py:252]: What is this case demonstrating?

File: music21/romanText/writeRoman.py [hunk 4]
```diff
@@ -255,27 +273,36 @@ def prepSequentialListOfLines(self):
                     msg = f'further time signature change(s) unprocessed: {unprocessedTSs}'
                     self.combinedList.append(f'Note: {msg}')
 
-            # RomanNumerals
-            measureString = ''  # Clear for each measure
+            # Initialise measure
+            thisMeasureNumber = thisMeasure.measureNumberWithSuffix()  # now str
```
Comment (mscuthbert) [inline music21/romanText/writeRoman.py:277]: typing.  Especially for anything called "...Number" that is actually a string.

File: music21/romanText/writeRoman.py [hunk 5]
```diff
@@ -255,27 +273,36 @@ def prepSequentialListOfLines(self):
...
+                if rn.tie is None or rn.tie.type == 'start':  # Ignore tied-to Roman numerals
                     chordString = self.getChordString(rn)
-                    measureString = rnString(measureNumber=thisMeasure.measureNumber,  # Suffix?
-                                             beat=rn.beat,
-                                             chordString=chordString,
-                                             inString=measureString,  # Creating update
-                                             )
+                    if rn.beat != 1:  # b1 not needed
+                        measureString += f' b{intBeat(rn.beat)}'
+                    measureString += f' {chordString}'
```
Comment (mscuthbert) [inline music21/romanText/writeRoman.py:292]: please explain the change here -- we're removing tested and testable code for a single RN and replacing it with nested code that is hard to test.  -1

File: music21/romanText/writeRoman.py [hunk 6]
```diff
@@ -303,53 +330,6 @@ def getChordString(self,
...
-    >>> lineStarter = romanText.writeRoman.rnString(14, 1, 'G: I')
-    >>> lineStarter
-    'm14 G: I'
-
-    For any other inString, that string is the start of a measure line continued by the new values
-
-    >>> continuation = romanText.writeRoman.rnString(14, 2, 'viio6', 'm14 G: I')
-    >>> continuation
-    'm14 G: I b2 viio6'
-
```
Comment (mscuthbert) [inline music21/romanText/writeRoman.py:325]: removal of well-explained, stable, and tested code for undocumented and untested routines.  negative.

File: music21/romanText/writeRoman.py [hunk 7]
```diff
@@ -542,19 +524,6 @@ def testTypeParses(self):
...
-        test = rnString(1, 1, 'G: I')
-        self.assertEqual(test, 'm1 G: I')  # no beat number given for b1
-
-        test = rnString(0, 4, 'b: V')
-        self.assertEqual(test, 'm0 b4 b: V')  # beat number is given for all other cases
-
-        with self.assertRaises(ValueError):  # error when the measure numbers don't match
-            rnString(15, 1, 'viio6', 'm14 G: I')
-
 # ------------------------------------------------------------------------------
```
Comment (mscuthbert) [inline music21/romanText/writeRoman.py:527]: removal of tests with no corresponding replacement.  -1

**Review by mscuthbert — CHANGES_REQUESTED** [review]
Please see comments on removing tested/documented code without replacements.

**Comment by MarkGotham** [comment]
Ok, noted.

The mass deletions are an attempt at simpler solutions and strike me as a good fit for the contributing docs' invitation to remove 30+ lines when 1 will do nicely. All the same, while I have misgiving about `rnString` (for that reason among others), I'll go ahead and reinstate it with the minimum changes to accommodate new tokens (i.e., repeat marks).

For reference, the process is about the addition of specific tokens in a specific order.
- measure (required)
- (optional) start repeat
- RNs (with beat except for b1, at least one required)
- (optional) end repeat

**Comment by MarkGotham** [comment]
This PR has been superseded by another, subsequent one for the same issue at #1435.

I wonder if there's an integration action for checking new PRs against those in progress that would prevent this?

In any case, I've been through to pick them apart and what remains as new here is simply:
- adding a doc test and
- one case of specifying `insert()` at position 0 (as requested) 
- misc clean up and clarification

A small contributions for sure, but a positive one, and makes this (1393) ready for closure.

Let me know if you spot any issues / anything missing.

File: music21/romanText/writeRoman.py (hunk 1 above)
Comment (mscuthbert) [reply music21/romanText/writeRoman.py:242]: This is still in the current version.  Unresolved.

File: music21/romanText/writeRoman.py [hunk 8]
```diff
@@ -236,17 +236,45 @@ def prepSequentialListOfLines(self):
...
+        >>> m2 = stream.Measure(number=2)
+        >>> m2.insert(0, meter.TimeSignature('3/4'))
+        >>> m2.leftBarline = bar.Repeat(direction='start')
+        >>> m2.rightBarline = bar.Repeat(direction='end')
+        >>> m2.insert(0, roman.RomanNumeral('I', 'G'))
+        >>> p.insert(0, m2)
+        >>> testCase = romanText.writeRoman.RnWriter(p)
+
+        The last line of the `.combinedList` gives the new measure:
+        >>> testCase.combinedList[-1]
```
Comment (mscuthbert) [inline music21/romanText/writeRoman.py:265]: Needs blank line between text and doctest.

File: music21/romanText/writeRoman.py [hunk 9]
```diff
@@ -236,17 +236,45 @@ def prepSequentialListOfLines(self):
...
+        'm2 ||: I :||'
+
+        The line before that gives the time signature change:
+        >>> testCase.combinedList[-2]
```
Comment (mscuthbert) [inline music21/romanText/writeRoman.py:269]: As above

**Review by mscuthbert — CHANGES_REQUESTED** [review]
The contributions here are enough to still merge after the changes requested are made.

If you know of an easy tool that can prevent duplicated work, happy to add it.

**Review by mscuthbert — APPROVED** [review]
LGTM.  Thanks!

**Pull request merged**